### PR TITLE
Email notification Feature Backend

### PR DESCRIPTION
**Pull request by AtharvaThorve**
#25 
This is the backend section for the Email notification system.<br>
Use the file `getEncPassword.js` to generate encrypted password and store it in `.env` file as `ENCPASSWORD=<generated enc password>` and also add `CRYPTOKEY` value as given in the `getEncPassword.js` file in the `.env` file.<br>
Also, change the email in `sendEmail.js`.

### CODE REVIEW

**Comment by AtharvaThorve** [comment]
Screenshot (27)
<br>
Screenshot (28)

Since a frontend is not implemented for this section request is made using the Rest Client extension on VSCode.

Also, the tests are failing since it requires some environment variables set to run them. I'll solve this error then it'll be good to be merged.

**Review by ShubhankarKG — COMMENTED** [review]
This looks great to me! Let's wait for @AliabbasMerchant to have a look.

File: server/src/controllers/Events.js
Comment (AliabbasMerchant) [inline server/src/controllers/Events.js:141]: nit: Try to extract out such repeated code into a function

File: server/src/utility/scheduler.js
Comment (AliabbasMerchant) [inline server/src/utility/scheduler.js:1]: Does schedule persist all scheduled jobs in a DB/file, or is it just in memory?
We need to make sure that it persists

File: server/src/utility/scheduler.js
Comment (AliabbasMerchant) [inline server/src/utility/scheduler.js:21]: I am not sure and a bit skeptical about this prefix, substring thing. Can we figure out a better way?

File: server/src/utility/sendEmail.js
Comment (AliabbasMerchant) [inline server/src/utility/sendEmail.js:17]: Please put such constants in a separate file

File: server/src/utility/scheduler.js
Comment (AtharvaThorve) [reply server/src/utility/scheduler.js:1]: In the documentation provided for the `node-schedule` package, it is given that jobs persist as long as the script is running.
> Note that Node Schedule is designed for in-process scheduling, i.e. scheduled jobs will only fire as long as your script is running, and the schedule will disappear when execution completes. If you need to schedule jobs that will persist even when your script isn't running, consider using actual cron.
> Documentation

File: server/src/controllers/Events.js
Comment (AtharvaThorve) [reply server/src/controllers/Events.js:7]: I did not understand this part.

File: server/src/utility/scheduler.js
Comment (AtharvaThorve) [reply server/src/utility/scheduler.js:21]: Yes, I'll look for other things.

File: server/src/controllers/Events.js
Comment (ShubhankarKG) [reply server/src/controllers/Events.js:7]: I assume that was for the cloudinary config details.

File: server/src/utility/scheduler.js
Comment (AliabbasMerchant) [reply server/src/utility/scheduler.js:1]: Okay.
But we need it to persist even after the script finishes/restarts. (Example scenario: server crashes / server is updated)

File: server/src/controllers/Events.js
Comment (AliabbasMerchant) [reply server/src/controllers/Events.js:7]: Yes

File: server/src/utility/scheduler.js
Comment (AtharvaThorve) [reply server/src/utility/scheduler.js:1]: For this, we can go for using mongodb, redis or static files.

Comment (AliabbasMerchant) [reply server/src/utility/scheduler.js:1]: Yes. But we will also have to read and schedule it again (after server starts), if the event does not exist.

Lets use MongoDB. We can refer to events by id, and schedule it again, if such an event does not exist.

Considering this is a big change, lets raise an issue for it, and do it in another PR. We can merge this PR once the other points are addressed

Comment (AtharvaThorve) [reply server/src/utility/scheduler.js:1]: We can do something like an user will contain an event id field for which he/she has added the reminder for.

Whenever the server restarts we can schedule the jobs using that field.

**Review by ShubhankarKG — APPROVED** [review]
Thanks for the changes, LGTM :fire:  Let's wait for @AliabbasMerchant to approve

File: server/src/utility/getMailOptions.js
Comment (AliabbasMerchant) [inline server/src/utility/getMailOptions.js:1]: Ideally this file should have been named `mailOptions.js`, but its okay

**Review by AliabbasMerchant — APPROVED** [review]

**Pull request merged**
